docs/internals/optimizer.rst

```diff
@@ -5,8 +5,10 @@
 The Optimizer
 *************
 
-The Solidity compiler uses two different optimizer modules: The "old" optimizer
-that operates at the opcode level and the "new" optimizer that operates on Yul IR code.
+The Solidity compiler involves optimizations at three different levels (in order of execution):
+- Optimizations in code generation based on a direct analysis of solidity code.
+- Optimizing transformations on the Yul IR code.
+- Optimizations on the opcode level.
```

Comment:
You need an empty line after lists in RST. Please check the rendered version to make sure the formatting is not broken.

While at it, I'd also suggest small tweaks:

```suggestion
The Solidity compiler involves optimizations at three different levels (in order of execution):

- Optimizations during code generation based on a direct analysis of Solidity code.
- Optimizing transformations on the Yul IR code.
- Optimizations at the opcode level.
```

Reply:
Done.